src/rgw/rgw_rest_s3.cc

```diff
@@ -3238,7 +3240,14 @@ int RGW_Auth_S3::authorize(RGWRados *store, struct req_state *s)
       if (algorithm != "AWS4-HMAC-SHA256") {
         return -EPERM;
       }
-      return authorize_v4(store, s);
+      /* compute first aws4 signature (stick to the boto2 implementation) */
+      int err = authorize_v4(store, s);
+      if (err && store->ctx()->_conf->rgw_s3_auth_aws4_presigned_url_bugfix) {
```

Comment:
yes, the current patch is a bit conservative. I tested with the new change and it looks working ok. can you confirm? thanks